### PR TITLE
0412jasna

### PR DESCRIPTION
**Pull request by jakrmpotic**
-Updated new tutorial to address comments from Dan/Will
-Moved new tutorial into Dev guide.
-Added python API installation info
-Addressed #133 re: removed specific Anaconda versions
-Restructured Dev guide - main changes =
 1.0  Installing Python API  
 2.0 Developing for Streaming Analytics (new tutorial)
 3.0 Developing for IBM Streams on-prem (previously creating your first app)
-Renumbered all sections

### CODE REVIEW

**Review by natashadsilva — COMMENTED** [review]
I think it needs to state explicitly what you need if you have Streams and what you need if you don't.   

@ddebrunner  If you do not have Streams installed, do you need the streamsx package AND the topology toolkit or will one or the other suffice?

Also, since these are options and not steps to be followed, they should not be numbered 1.1 and 1.2 but presented as either/or.  It might even be possible to merge this section into the introduction.

**Comment by natashadsilva** [comment]
I notice that the ability to develop Python applications without requiring Streams is not mentioned until section 2.0.  This is an important feature and should be highlighted in the introduction.  
Also, the introduction to the dev guide talks about how you can run your application (STANDALONE/DISTRIBUTED/etc) and then  mentions what you can do with the API.  I think the order of these two sections needs to be reversed.
I think the guide should talk about what the API allows, your options for creating applications using the API (with/without local install) and then your options for running the application.

File: docs/python/1.6/python-appapi-devguide-1.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+   ```
+   python
+   ```
+
+2. Go to your service dashboard, click **Service Credentials**, and then **View Credentials** to copy your service credentials.
+
+   Paste your credentials using the following command:
+
+   ```python
```
Comment (natashadsilva) [inline docs/python/1.6/python-appapi-devguide-1.md]: This section appears with the three quotes (```)  in the snippet.

File: docs/python/1.6/python-appapi-devguide-2.md [hunk 2]
```diff
@@ -9,43 +9,21 @@ prev:
   file: python-appapi-devguide
   title: Developing IBM Streams Applications with Python
 next:
-  file: python-appapi-devguide-3
-  title: 2.0 Developing your first application
+  file: python-appapi-devguide-2a
+  title: 2.0 Developing for the Streaming Analytics service
```
Comment (natashadsilva) [inline docs/python/1.6/python-appapi-devguide-2.md]: The titles for the two sections are a bit incorrect.
"Developing for the Streaming Analytics service" and "for Streams on-premises" make it sound mutually exclusive.  Really, though, the core applications themselves developed in the same way.  The difference is how you submit them and how you configure your environment.  So perhaps "developing without installing streams"  and "developing with a streams install" work better.

**Comment by natashadsilva** [comment]
I also noticed that the hello world tutorial is very limited in comparison to the temperature sensors tutorial (compare section 2.3 with sections 3.2 to 3.6).   I propose that we use the same application. I changed the bluemix tutorial to re-use the on-prem tutorial:
I have done this on my fork and you can take a look here:
https://natashadsilva.github.io/streamsx.documentation/docs/python/1.6/python-appapi-devguide-2/
I worked through the tutorial step by step and it works as expected. If you think this is okay then I will create a pull request.

**Comment by jakrmpotic** [comment]
@natashadsilva Re: you comment
"
I think it needs to state explicitly what you need if you have Streams and what you need if you don't.
@ddebrunner If you do not have Streams installed, do you need the streamsx package AND the topology toolkit or will one or the other suffice?
Also, since these are options and not steps to be followed, they should not be numbered 1.1 and 1.2 but presented as either/or. It might even be possible to merge this section into the introduction.
"

OK. clarify that, If you do not have Streams installed, you need EITHER the toolkit or they Python package. They  have the came content. 
I'll remove the section numbers and just make them definition lists. We'll keep the installation info in it's own topic for easier retrieval ( the link on the nav).

**Comment by jakrmpotic** [comment]
@natashadsilva  re: your comment:
"I notice that the ability to develop Python applications without requiring Streams is not mentioned until section 2.0. This is an important feature and should be highlighted in the introduction.
Also, the introduction to the dev guide talks about how you can run your application (STANDALONE/DISTRIBUTED/etc) and then mentions what you can do with the API. I think the order of these two sections needs to be reversed.
I think the guide should talk about what the API allows, your options for creating applications using the API (with/without local install) and then your options for running the application."

I'll update the main topic after I merge #129. Am waiting for @wmarshall484 to finish making updates to the topic.

File: docs/python/1.6/python-appapi-devguide-2a.md [hunk 3]
```diff
@@ -0,0 +1,275 @@
+---
+layout: docs
+title:  2.0 Developing without IBM Streams on-premises
```
Comment (ddebrunner) [inline docs/python/1.6/python-appapi-devguide-2a.md]: ~~without IBM Streams on-premises~~
without an IBM Streams install

Actually, have a positive statement, - Developing for IBM Streaming Analytics

File: docs/python/1.6/python-appapi-devguide-2a.md [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+
+## 2.1 Setting up your Python environment
+
+The following steps show how you can set up your Python development environment:
+
+1. Ensure that you have Java 8 installed, and the JAVA_HOME environment variable is set:
+
+   ```
+   export JAVA_HOME="/usr/lib/jvm/java-1.x.x-openjdk"
+   export PATH=$JAVA_HOME/bin:$PATH
```
Comment (ddebrunner) [inline docs/python/1.6/python-appapi-devguide-2a.md]: ~~export PATH=$JAVA_HOME/bin:$PATH~~

Not required.

File: docs/python/1.6/python-appapi-devguide-3.md [hunk 5]
```diff
@@ -19,8 +19,46 @@ The application must also be scalable. If the number of temperature sensors doub
...
+
+   * *Recommended* - Anaconda with Python 3.5 [https://www.continuum.io/downloads](https://www.continuum.io/downloads).
+
+   * CPython 3.5 or later [https://www.python.org](https://www.python.org).
+
+   The Python Application API was tested with Python 3.5.1.
+
+   **Important:** To build IBM Streams application bundles with the Python Application API that can be submitted to your IBM Streaming Analytics service:
+     * You must use Python 3.5
+     * Install Anaconda at `/disk1/opt/Anaconda3` on the machine where you execute the Python code that builds the topology and submits it to the `BUNDLE` context.
```
Comment (ddebrunner) [inline docs/python/1.6/python-appapi-devguide-3.md]: Remove - this requirement (/disk1/opt/Anaconda3) was removed.

File: docs/python/1.6/python-appapi-devguide-3.md [hunk 6]
```diff
@@ -19,8 +19,46 @@ The application must also be scalable. If the number of temperature sensors doub
...
+        source /home/streamsadmin/InfoSphere_Streams/4.2.0.0/bin/streamsprofile.sh
+
+
+1. If you are using IBM Streams 4.2 or later, you can skip this step because the Python Application API is included at* `$STREAMS_INSTALL/toolkits/com.ibm.streamsx.topology`. <br><br>If you are using an earlier version of IBM Streams, you must:
+    1. Download the latest version of the IBM Streams Topology toolkit from the IBMStreams organization on GitHub from the streamsx.topology [Releases page](https://github.com/Ibmstreams/streamsx.topology/releases).
+    1. After the toolkit downloads, extract it to your file system.
+
+1. Install a supported version of Python:
+
+   * *Recommended* - Anaconda with Python 3.5 [https://www.continuum.io/downloads](https://www.continuum.io/downloads).
```
Comment (ddebrunner) [inline docs/python/1.6/python-appapi-devguide-3.md]: 2.7 or 3.5

File: docs/python/1.6/python-appapi-devguide-3.md [hunk 7]
```diff
@@ -19,8 +19,46 @@ The application must also be scalable. If the number of temperature sensors doub
...
+
+   * CPython 3.5 or later [https://www.python.org](https://www.python.org).
```
Comment (ddebrunner) [inline docs/python/1.6/python-appapi-devguide-3.md]: CPython 2.7 or 3.5

(remove 'or later')

File: docs/python/1.6/python-appapi-devguide-3.md [hunk 8]
```diff
@@ -19,8 +19,46 @@ The application must also be scalable. If the number of temperature sensors doub
...
+
+   The Python Application API was tested with Python 3.5.1.
```
Comment (ddebrunner) [inline docs/python/1.6/python-appapi-devguide-3.md]: Remove this line.

File: docs/python/1.6/python-appapi-devguide-3.md [hunk 9]
```diff
@@ -101,11 +132,11 @@ streamsx.topology.context.submit("STANDALONE", topo)
 **Remember:** You can run your application in the following ways:
 
 * As a **Streams distributed application** (DISTRIBUTED). When running in this mode, the application produced will be deployed automatically on your IBM Streams instance.
```
Comment (ddebrunner) [inline docs/python/1.6/python-appapi-devguide-3.md:137]: Missing 'STREAMING_ANALYTICS_CONTEXT

File: docs/python/1.6/python-appapi-devguide-2a.md [hunk 10]
```diff
@@ -0,0 +1,275 @@
...
+
+A streaming analytics application is a directed flow graph that specifies how data is generated and processed. The `Topology` object contains information about the structure of the directed flow graph.
+
+
+## 2.5 Defining a data source
+The `Topology` object also includes functions that enable you to define your data sources. In this application, the data source is the temperature sensor.
+
+In this example, simulate temperature sensor readings by defining a Python generator function that returns an iterator of random numbers.
+
+**Important:** Callable inputs to functions, such as the definition of the `readings()` function, cannot be defined in the `'_main_'` module. The inputs must be defined in a separate module.
```
Comment (ddebrunner) [inline docs/python/1.6/python-appapi-devguide-2a.md]: Remove this "Important" section. It no longer applies.

File: docs/python/1.6/python-appapi-devguide-2a.md (hunk 3 above)
Comment (jakrmpotic) [reply docs/python/1.6/python-appapi-devguide-2a.md]: OK. will change back.

File: docs/python/1.6/python-appapi-devguide-2a.md (hunk 4 above)
Comment (jakrmpotic) [reply docs/python/1.6/python-appapi-devguide-2a.md]: Removed

File: docs/python/1.6/python-appapi-devguide-3.md (hunk 5 above)
Comment (jakrmpotic) [reply docs/python/1.6/python-appapi-devguide-3.md]: @ddebrunner pls confirm - I can remove the whole note labelled important?

Comment (jakrmpotic) [reply docs/python/1.6/python-appapi-devguide-3.md]: actually, I will keep the part that 3.5 is required. Will remove just the second bullet.

File: docs/python/1.6/python-appapi-devguide-3.md (hunk 6 above)
Comment (jakrmpotic) [reply docs/python/1.6/python-appapi-devguide-3.md]: revised.

File: docs/python/1.6/python-appapi-devguide-3.md (hunk 7 above)
Comment (jakrmpotic) [reply docs/python/1.6/python-appapi-devguide-3.md]: Revised: 
screen shot 2017-04-13 at 3 33 51 pm

Comment (jakrmpotic) [reply docs/python/1.6/python-appapi-devguide-3.md]: Will remove the line re: app was tested with 3.5.1

File: docs/python/1.6/python-appapi-devguide-3.md (hunk 9 above)
Comment (jakrmpotic) [reply docs/python/1.6/python-appapi-devguide-3.md:137]: Added.

File: docs/python/1.6/python-appapi-devguide-2a.md (hunk 10 above)
Comment (jakrmpotic) [reply docs/python/1.6/python-appapi-devguide-2a.md]: Done.

**Pull request merged**